tests/main/validate-container-happy/task.yaml

```diff
@@ -0,0 +1,43 @@
+summary: check the symlinks following the right track
+
+environment:
+    SNAP: test-snapd-validate-container-happy
+    SNAP_INSTALL_DIR: $(pwd)/snap-install-dir
+    SNAP_MOUNT_DIR: $(os.paths snap-mount-dir)
+
+prepare: |
+    snap pack test-snapd-validate-container-happy
+
+execute: |
+
+    # We shouldn't use relative symlinks in Github as they cannot be packed correctly.
+    # So here let's test whether we can still pack such symlinks within a snap and use if needed.
+    # First we "try" to unpack the snap structure and untar the symlinks
+    # Then we pack the snap with these symlinks and then install
+    # Finally we check to see if the symlinks actually support the intervined symlinks
+
+    # Untar the symlinks 
+    mkdir -p "$SNAP_MOUNT_DIR"/"$SNAP"/hell
+    tar -xvf "$SNAP"/hell/hell.tar -C "$SNAP_MOUNT_DIR"/"$SNAP"/hell
+
+    snap try "$SNAP"
+    # Check to see if the symlinks exist
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/bar | MATCH "foo -> baz -> qux"
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/baz | MATCH "foo -> bar -> qux"
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/foo | MATCH "bar -> baz -> qux"
+    readlink "$SNAP_MOUNT_DIR"/"$SNAP"/hell/qux | MATCH "foo -> bar -> baz"
+
+    # Create a new snap structure that includes the unpacked symlinks
+    mkdir -p "$SNAP_INSTALL_DIR"
+    cp -ra "$SNAP"/* "$SNAP_INSTALL_DIR"
+    cp -ra "$SNAP_MOUNT_DIR"/"$SNAP"/hell "$SNAP_INSTALL_DIR"
+
+    snap pack "$SNAP_INSTALL_DIR"
+    snap install --dangerous test-snapd-validate-container-happy_1.0_all.snap
```

Comment:
Ideally we would remove the snap that we installed in try mode with `snap try` before installing this .snap file, since both are for the same snap and we essentially end up doing a refresh which is more complicated of a test than we need for here, so my preference would be to do:

1. unpack tarball
2. snap try the dir
3. check symlinks exist in /snap/...
4. remove snap
5. snap pack the dir
6. snap install the .snap file
7. check symlinks exist in /snap/...

does that make sense?

Reply:
Simplified the process accordingly. Current steps:

1. Untar the symlinks to the existing folder
2. "try" the folder as a new snap
3. Check that symlinks point to the right target
4. Remove the snap
5. Pack the directory as a new snap
6. Install the snap
7. Check that symlinks point to the right target
8. Remove the snap